### DISCUSSION TITLE
feature: open up dashboard on last buffer deleted (closed).

### DISCUSSION THREAD

**Discussion by hifanx** [folke/snacks.nvim#663] [Q&A]
### Did you check the docs?

- [X] I have read all the snacks.nvim docs

### Is your feature request related to a problem? Please describe.

Switching from Alpha because I'm minimalizing my setup.
When using alpha I can configure to open up dashboard when the last buffer is deleted.
I would request the same feature to be available with snacks.

### Describe the solution you'd like

There's a (half) workaround when using Alpha that, combined with bufdelete.nvim, which comes with a `BDeletePre *` pattern, can allow us to open up Alpha upon last buffer closed with a autocmd.  Please refer to [this issue](https://github.com/goolord/alpha-nvim/discussions/85) for details. 

It is by no means a convenient or functionally perfect solution.

It would be best that there is a config entry that, when set to true, can open up the dashboard automatically when the last buffer is deleted while preserving any layout, e.g. all kinds of trees, trouble, etc, if any.

### Describe alternatives you've considered

This enhancement can happen with improving the native logic of snacks' bufdelete.
When calling Snacks.bufdelete(), if it's the last buffer, open up dashboard, If dashboard is already open, DO NOTHING.

**Comment by ItsLiyua** [comment]
I'd be interested in this as well. I tried my best with autocmds but couldn't get to to work.

**Comment by nexfader** [comment]
I've solved it by wrapping Snacks.bufdelete:
```lua
local open_dashboard_if_empty_buffer = function()
  local buf = vim.api.nvim_get_current_buf()
  local win = vim.api.nvim_get_current_win()
  local is_empty = vim.api.nvim_buf_get_name(buf) == "" and vim.bo[buf].filetype == '' and not vim.bo[buf].modified
  if not is_empty then return end

  require('snacks.dashboard').open({ buf = buf, win = win })
end

_G.close_buffer = function()
  require('snacks').bufdelete({
    filter = function(buf)
      return vim.bo[buf].filetype ~= 'snacks_dashboard'
    end
  })

  vim.schedule(open_dashboard_if_empty_buffer)
end
```

Usage:
```lua
vim.keymap.set('n', '<leader>bd', close_buffer, { desc = 'Kill buffer' })
```
